addons/web/static/tests/core/utils/hooks.test.js

```diff
@@ -310,10 +310,63 @@ describe("useService", () => {
                 useService("toy_service");
             }
         }
-
+        const originalHandleError = window.owl.App.prototype.handleError;
+        // Catch the error thrown by the owl and let the runner do its job
+        window.owl.App.prototype.handleError = function (...args) {
+            try {
+                patchWithCleanup(console, { warn: () => {} });
+                return originalHandleError(...args);
+            } catch (error) {
+                console.log(`This error is caught by the runner of the test: ${error.message}`);
+            }
+        };
         await expect(mountWithCleanup(MyComponent)).rejects.toThrow(
-            "Service toy_service is not available"
+            'Service toy_service is not available in "MyComponent".'
         );
+        window.owl.App.prototype.handleError = originalHandleError;
+    });
+
+    test("service becomes available after a delay", async () => {
+        useServiceProtectMethodHandling.fn = useServiceProtectMethodHandling.original;
+        let myComponent;
+        class MyComponent extends Component {
+            static template = xml`<div class="o-MyComponent"><div t-if="myService" t-att-class="{
+                'o-MyComponent-serviceAvailable': myService.someAttr === 1,
+                'o-MyComponent-serviceReactivelyAvailable': myService.someAttr === 2,
```

Comment:
I would put comment to give more context on the test, to give motivation this is valuable and we want it.
Here's a rough example:

```js
// This test illustrates case with lazy-loading of bundle that contains both main components and services.
// Services are available become (main) components at page load.
// In lazy-loading, there's no single flow to ensure new services are deployed before (main) components.
// so new (main) component can very temporarily be setup when service is not yet deployed.
// `useService()` is a component hook so this has to handle slight asynchronous deployment of service.
```